tools/optimizer/optimizer.cpp

```diff
@@ -716,7 +716,7 @@ Ref simplifyCondition(Ref node) {
 
 StringSet ELIMINATION_SAFE_NODES("assign call if toplevel do return label switch binary unary-prefix"); // do is checked carefully, however
 StringSet IGNORABLE_ELIMINATOR_SCAN_NODES("num toplevel string break continue dot"); // dot can only be STRING_TABLE.*
-StringSet ABORTING_ELIMINATOR_SCAN_NODES("new object function defun for while array throw"); // we could handle some of these, TODO, but nontrivial (e.g. for while, the condition is hit multiple times after the body)
+StringSet ABORTING_ELIMINATOR_SCAN_NODES("new object defun while array"); // we could handle some of these, TODO, but nontrivial (e.g. for while, the condition is hit multiple times after the body)
```

Comment:
Well in that case this check can go as well, right? https://github.com/kripken/emscripten/blob/58f1de90c85a1b814d7ce7a7478de24564de8bfa/tools/optimizer/optimizer.cpp#L2411

If you're confident then I'll remove them (I agree that defun can go but the others make me nervous).